elytron/ELY-2538-caching-realm-update-cached-credential.adoc

```diff
@@ -0,0 +1,80 @@
+= ELY-2538 Provide a possibility for a caching realm to authenticate users with underlying realm when credential verification with cached credential fails
+:author:            Diana Krepinska
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+For LDAP servers that do not support persistent search, we do not receive notifications if the cached identities have had their passwords changed in LDAP server externally.
+
+This issue is to provide a possibility for the caching realm to authenticate a user with the underlying realm directly when the authentication fails with the cached credential. If the credential verification is successful with the underlying realm, remove the old credential from cache and put the updated credential to the cache.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/ELY-2538[ELY-2538]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-2069[EAP7-2069] - EAP issue tracker
+* https://issues.redhat.com/browse/ELY-2538[ELY-2538] - Elytron project issue tracker
+* https://issues.redhat.com/browse/WFLY-18185[WFLY-18185] - Wildfly community documentation issue tracker
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Krepinska]
+
+=== QE Contacts
+
+* mailto:TODO
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron project]
+* Wildfly Security component will be consequently affected by the improvements introduced by this new feature
+
+== Requirements
+
+=== Hard Requirements
+
+* It must be possible for the caching realm to authenticate a user and update the cached credential if it has been changed externally and a user attempts to authenticate with the new credential.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+This functionality should either be added as a default, as this was the behaviour of the legacy security subsystem. Or we can add this behaviour under a new attribute of the caching realm. If we add it as configurable with a property, then it can be backported with a system property.
+
+I have decided to enable this functionality by default. The reason for this is that this functionality was present in the legacy security, and users can consider the absence of this functionality to be a bug, so we should not require its configuration.
```

Comment:
You mention enable by default but it does not look now like there will be a mechanism to turn it off?  May be worth stating it is not optional if that is the case.